app/src/main/java/br/com/alexf/boraprofut/features/timer/TimerUserCase.kt

```diff
@@ -0,0 +1,41 @@
+package br.com.alexf.boraprofut.features.timer
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class TimerUserCase {
+
+    private var job: Job = Job()
+    var isPause = true
+    private var currentTimeMillis = 0L
+    var timeMillis = 0L
+        set(value) {
+            field = value
+            currentTimeMillis = field
+        }
+    private var _timer = MutableStateFlow(currentTimeMillis)
+    var timer = _timer.asStateFlow()
+
+    fun startTimer() {
+        job = CoroutineScope(IO).launch {
+            isPause = false
+            while (currentTimeMillis > 0) {
+                delay(1000)
+                if (isPause) {
+                    continue
```

Comment:
Com o continue, o loop será mantido enquanto o usuario fica na tela.
Poderíamos substituir por break para de fato parar o loop e na viewmodel , na fun resume, adicionar novamente a chamada de startTime para reinicia ro loop de onde parou
```suggestion
                    break
```